### PR TITLE
Add Sentries to try to understand what's happening

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt [hunk 1]
```diff
@@ -168,6 +168,7 @@ object SentryDebug {
     fun sendAlreadyExistingMessage(folder: Folder, message: Message, threadMode: ThreadMode) {
         Sentry.withScope { scope ->
             scope.level = SentryLevel.ERROR
+            scope.setExtra("folder.cursor", folder.cursor ?: "null")
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:171]:
```suggestion
            scope.setExtra("folder.cursor", "${folder.cursor}")
```

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt [hunk 2]
```diff
@@ -393,6 +396,7 @@ class RefreshController @Inject constructor(
 
         val impactedThreads = mutableSetOf<Thread>()
         val paginationInfo = getPaginationInfo()
+        addSentryBreadcrumbAboutPaginationInfo(info = paginationInfo)
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt:399]:
```suggestion
        addSentryBreadcrumbAboutPaginationInfo(info = paginationInfo)

```

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt:399]: I'll do differently.

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt [hunk 3]
```diff
@@ -652,12 +655,13 @@ class RefreshController @Inject constructor(
...
         val impactedThreadsManaged = mutableSetOf<Thread>()
         val folderMessages = folder.messages.associateByTo(mutableMapOf()) { it.uid }
+        val addedMessagesUids = mutableListOf<Int>()
 
         remoteMessages.forEach { remoteMessage ->
             scope.ensureActive()
 
             val shouldSkipThisMessage = addRemoteMessageToFolder(remoteMessage, folder, folderMessages)
-            if (shouldSkipThisMessage) return@forEach
+            if (shouldSkipThisMessage) return@forEach else addedMessagesUids.add(remoteMessage.shortUid)
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt:664]: No else after return

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by FabianDevel — APPROVED** [review]

**Pull request merged**